central/apitoken/service/service_impl.go

```diff
@@ -101,7 +101,7 @@ func (s *serviceImpl) GenerateToken(ctx context.Context, req *v1.GenerateTokenRe
 		return nil, err
 	}
 	if err := verifyNoPrivilegeEscalation(id.Roles(), roles); err != nil {
-		return nil, errox.NewErrNotAuthorized(err.Error())
+		return nil, errox.NotAuthorized.CausedBy(err.Error())
```

Comment:
```suggestion
		return nil, errox.NotAuthorized.CausedBy(err)
```

Reply:
@0x656b694d I would do this in separate PR.